GNUmakefile

```diff
@@ -37,7 +37,7 @@ test: fmtcheck
 .PHONY: testacc
 testacc: fmtcheck
 	@$(eval VERSION=acc)
-	TF_ACC=1 go test $(TEST) -run '$(TEST_REGEX)' -v -parallel 5 $(TESTARGS) -timeout $(ACCTEST_TIMEOUT) -cover -ldflags="$(LINKER_FLAGS)"
+	TF_ACC=1 go test $(TEST) -run '$(TEST_REGEX)' -v -parallel 20 $(TESTARGS) -timeout $(ACCTEST_TIMEOUT) -cover -ldflags="$(LINKER_FLAGS)"
```

Comment:
`-parallel` should not exceed the number of cores, do you know how many cores you have in GH?

Reply:
My understanding is that `go test` uses `GOMAXPROCS` (which is set to the number of CPUs) as the default value when the `-parallel` flag is not specified. I'm not currently aware of any recommendation advocating for using this default value. If you happen to have come across an article or resource that suggests this, could you share it with me?

Reply:
I mean yes, parallel defaults to `GOMAXPROCS` but the flag is more to lower it than increase it, there's no benefit to bump it above the number of available cores since each parallel test will be in contention for resources and not really running in prallel

Reply:
The ubuntu-latest seems to have only 2 cores [supported-runners-and-hardware-resources](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources). I will try to remove the parallel flag and see if the test has the same behaviour. I will add an update here. Thanks!

Reply:
I tested here [9401709790](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/5210481259/jobs/9401709790). When the `-parallel` flag is not provided, it runs only 2 acceptance tests in parallel (2 projects available in cloud-dev)
![Screenshot 2023-06-08 at 12 12 35](https://github.com/mongodb/terraform-provider-mongodbatlas/assets/5663078/ce8506af-4b42-4a17-b97a-7b4ad737d428)


When the flag is provided, it was running all of them in parallel (4 acceptance tests -> 4 projects in cloud-dev)
![Screenshot 2023-06-07 at 16 23 37](https://github.com/mongodb/terraform-provider-mongodbatlas/assets/5663078/c6c549ea-e0e4-430d-9a74-43cd3b46dd85)

Reply:
there's a difference between parallelism and concurrency, I can't see the logs for [9401709790](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/5210481259/jobs/9401709790). so can't confirm yet but considering concurrent vs parallel what go is doing is spinning `--parallel` processes if the number of processes exceed the number of CPUs what will happen is go will have to pause some tests to run other tests, you will see this on the logs as PAUSE/CONT statements so they run concurrently, if the times are better with  `-parallel 20` than with 4 or other number feel free to keep but 20 is not really 20 parallel test but 10 concurrent test (assuming 2 CPU)

Reply:
Thanks for the clarification! Yes, the time is better with `--parallel 20` as the job took 24 minutes [5201802898](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/5201802898) vs 47 minutes [5210481259](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/5210481259?pr=1219)